src/extension.ts

```diff
@@ -885,3 +887,33 @@ function isSupportedLanguage(languageId: TextDocument["languageId"]): boolean {
       return false;
   }
 }
+
+// NOTE(gabro): we would normally use the `configurationDefaults` contribution point in the
+// extension manifest but that's currently limited to language-scoped settings in VSCode.
+// We use this method to set global configuration settings such as `files.watcherExclude`
+// and `search.exclude`.
+function setupMetalsSettings() {
+  function updateFileConfig(
+    configKey: string,
+    propertyKey: string,
+    newValues: Record<string, boolean>
+  ) {
+    const config = workspace.getConfiguration(configKey);
+    const currentValues = (config.get(propertyKey) || {}) as Record<
+      string,
+      boolean
+    >;
+    config.update(
+      propertyKey,
+      { ...currentValues, ...newValues },
+      ConfigurationTarget.Global
+    );
+  }
+  const ignoredMetalsDirs = {
+    "**/.bloop": true,
+    "**/.metals": true,
+    "**/target": true,
```

Comment:
This should cover sbt, is there any other generated directory from other build tools we should include?

Reply:
`"**/.ammonite"`?

Reply:
We could also exclude `out` directory - I think that is used for mill. We could add an if workspace contains `build.sc` only.